gnovm/pkg/gnolang/preprocess.go

```diff
@@ -426,6 +427,24 @@ func Preprocess(store Store, ctx BlockNode, n Node) Node {
 			switch n := n.(type) {
 			// TRANS_ENTER -----------------------
 			case *AssignStmt:
+				if n.Op == DEFINE {
+					for _, lx := range n.Lhs {
+						ln := lx.(*NameExpr).Name
+						if ln == blankIdentifier {
+							// ignore.
+						} else if strings.HasPrefix(string(ln), ".tmp_") {
```

Comment:
Can we call these variables differently? I'd suggest `.decompose_%d_%d`. I'd also suggest moving checking whether it's a tmp variable, and generating the name itself, to two specific functions separated from Preprocess

Reply:
@thehowl agree `.decompose_%d_%d` is a better name. I will update it. As for separate functions, we can  do that when there are more cases sharing the same logic.